lib/node_modules/@stdlib/array/base/map/README.md

```diff
@@ -0,0 +1,189 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2025 The Stdlib Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# map
+
+> Apply a callback function to elements in an input array and assign results to elements in a new output array.
+
+<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->
+
+<section class="intro">
+
+</section>
+
+<!-- /.intro -->
+
+<!-- Package usage documentation. -->
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var map = require( '@stdlib/array/base/map' );
+```
+
+#### map( x, fcn\[, thisArg] )
+
+Applies a callback function to elements in an input array and assigns results to elements in a new output array.
+
+```javascript
+var naryFunction = require( '@stdlib/utils/nary-function' );
+var abs = require( '@stdlib/math/base/special/abs' );
+
+var x = [ -1.0, -2.0, -3.0, -4.0 ];
+
+var y = map( x, naryFunction( abs, 1 ) );
+// returns [ 1.0, 2.0, 3.0, 4.0 ]
+```
+
+The function accepts the following arguments:
+
+-   **x**: input array.
+-   **fcn**: callback function.
+-   **thisArg**: callback execution context (_optional_).
+
+To set the callback function's execution context, provide a `thisArg`.
+
+<!-- eslint-disable no-invalid-this -->
+
+```javascript
+function count( x ) {
+    this.count += 1;
+    return x;
+}
+
+var x = [ 1.0, 2.0, 3.0, 4.0 ];
+
+var ctx = {
+    'count': 0
+};
+
+var y = map( x, count, ctx );
+// returns [ 1.0, 2.0, 3.0, 4.0 ]
+
+var v = ctx.count;
+// returns 4
+```
+
+The callback function is provided the following arguments:
+
+-   **value**: current array element.
+-   **index**: current element index.
+-   **arr**: input array.
+
+#### map.assign( x, y, fcn\[, thisArg] )
```

Comment:
@headlessNode You need to support a stride and offset parameter. See, e.g., https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/base/cuevery-by#cueverybyassign-x-out-stride-offset-predicate-thisarg--.